### DISCUSSION TITLE
Using the validation stream

### DISCUSSION THREAD

**Discussion by shahariel** [ContinualAI/avalanche#1374] [Q&A] [answered]
Hi!
I saw in some of the experiments examples the use of "benchmark_with_validation_stream" like this:
`scenario = benchmark_with_validation_stream(scenario, 0.05)`
And then inside the training loop:
`cl_strategy.eval(scenario.test_stream[: t + 1])`
(For example [here](https://github.com/ContinualAI/continual-learning-baselines/blob/main/experiments/split_mnist/online_replay.py) and [here](https://github.com/ContinualAI/continual-learning-baselines/blob/main/experiments/split_cifar100/replay.py)).

Is there a reason that after training an experiment you don't use the validation stream for evaluation? Is it in use some place else? If so, where?

Thank you,
Shahar

**Comment by AntonioCarta** [comment, marked as answer]
Hi, I believe the validation stream was used only for model selection, which we don't do in the examples.

**Reply by shahariel** [reply]
Thank you!
